### PR TITLE
initial COMPILING doc with linux, win, mac (new)

### PR DESCRIPTION
**Pull request by kirbysayshi**
As mentioned in #438, I figured it would be good to have documentation on how to compile close to the source. This document definitely ended up being longer than I anticipated, so if people have suggestions on how to shorten it, please let me know.

I suppose one option is to fold the new OS X information into the wiki page, and just put bare minimum info in the file with links to the wiki.

Curious what people think and would like. Please let me know!

### CODE REVIEW

File: COMPILING.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+- Build the solution.
+- Built files can be found in `build/<project name>/<build configuration>/`.
+
+Linux
+=====
+
+Install Dependencies
+--------------------
+
+	sudo apt-get install build-essential cmake git libopenal-dev zlib1g-dev libpng12-dev
```
Comment (RostakaGmfun) [inline COMPILING.md]: OpenGL development libraries should be added as well

File: COMPILING.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+Adding the PPA on Ubuntu 12.04 LTS, 12.10, and 13.04:
+
+	sudo add-apt-repository ppa:mir-team/staging
+	sudo apt-get update
+
+Installing:
+
+	sudo apt-get install libsdl2-dev
+
+More instructions can be found on the [project wiki][], such as how to compile from sdl2 from source.
```
Comment (RostakaGmfun) [inline COMPILING.md]: Redundant from

File: COMPILING.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+Configure Build (make)
+----------------------
+
+	mkdir build
+	cd build
+	cmake -G 'Unix Makefiles' ..
```
Comment (RostakaGmfun) [inline COMPILING.md]: `Unix Makefiles` is default on Linux

**Comment by xycaleth** [comment]
Closing due to inactivity. Feel free to resubmit the pull request again :)

**Pull request closed without merging**